tutor/src/screens/assignment-grade/index.js

```diff
@@ -77,12 +78,25 @@ class AssignmentGrading extends React.Component {
               onChange={ux.setSelectedPeriod}
             />
           </Heading>
-          <QuestionsBar ux={ux} />
-          <BodyBackground>
-            <Question
-              ux={ux}
-            />
-          </BodyBackground>
+          {
+            ux.headings.length === 0 
+              ? <>
+                <h2>No activity yet</h2>
+                <p>
+                No students enrolled.
+                Manage student enrollment in <TutorLink to="courseRoster" params={{ courseId: ux.course.id }}>Course roster</TutorLink>.
+                </p>
+              </>
+              : <>
+                <QuestionsBar ux={ux} />
+                <BodyBackground>
+                  <Question
+                    ux={ux}
+                  />
+                </BodyBackground>
+              </>
+          }
+          
```

Comment:
hm, rather than the big ternary, how about moving each block to either a component or a method, and then:
`ux.headings.length === 0 ? this.renderEmpty() : this.renderContent()`

Reply:
I was just doing that :) thanks nathan.